build.go

```diff
@@ -38,6 +39,11 @@ type PythonInstaller interface {
 	) error
 }
 
+// PythonPipCleanup defines the interface for cleaning up pip after a python installation
+type PythonPipCleanup interface {
+	Cleanup(packages []string, targetLayer string) error
+}
+
```

Comment:
Does "cleaning up" pip means deleting `pip`? If so it wouldn't be a good idea for workspaces as we need `pip`